sweep/sweeper.go

```diff
@@ -832,31 +829,19 @@ func (s *UtxoSweeper) sweep(set InputSet) error {
 	return nil
 }
 
-// markInputsPendingPublish saves the sweeping tx to db and updates the pending
-// inputs with the given tx inputs. It also increments the `publishAttempts`.
-func (s *UtxoSweeper) markInputsPendingPublish(tr *TxRecord,
-	inputs []*wire.TxIn) error {
-
-	// Add tx to db before publication, so that we will always know that a
-	// spend by this tx is ours. Otherwise if the publish doesn't return,
-	// but did publish, we'd lose track of this tx. Even republication on
-	// startup doesn't prevent this, because that call returns a double
-	// spend error then and would also not add the hash to the store.
-	err := s.cfg.Store.StoreTx(tr)
-	if err != nil {
-		return fmt.Errorf("store tx: %w", err)
-	}
-
+// markInputsPendingPublish updates the pending inputs with the given tx
+// inputs. It also increments the `publishAttempts`.
+func (s *UtxoSweeper) markInputsPendingPublish(set InputSet) {
 	// Reschedule sweep.
-	for _, input := range inputs {
-		pi, ok := s.pendingInputs[input.PreviousOutPoint]
+	for _, input := range set.Inputs() {
+		pi, ok := s.pendingInputs[*input.OutPoint()]
```

Comment:
yeah agreed. I'll try to change it in the following PR to avoid rebase conflicts, as this `OutPoint()` method is used in many existing places already.

Reply:
fixed in https://github.com/lightningnetwork/lnd/pull/8581/commits/ee03cd5e71e164372fc5780cefb3d12d1794b113